### PR TITLE
Add support for Symfony 4.3

### PR DESCRIPTION
**Pull request by epinxteren**
- Add symfony/phpunit-bridge to make sure we are not using deprecated
dependencies
- Add extra CI run for SF 4 with php 7.2

https://www.pivotaltracker.com/story/show/167697164

### CODE REVIEW

**Comment by jorissteyn** [comment]
Hi Eric,

The `MetadataFactory` service depends on `@templating` - which is deprecated in symfony 4.3 and no longer loaded by default. In order for this bundle to work in symfony 4.3 you must first require `symfony/templating` and your framework config should read:

```
framework:
    templating:
        engines:
        - twig
```

Perhaps the README.md should mention this. In the future, to support symfony 5, the MetadataFactory should use a `TwigEngine` instead of a `EngineInterface`.

**Comment by epinxteren** [comment]
@jorissteyn Hi Joris, Added documentation. For now i have only done the most minimal adjustment to support sf4.

File: composer.json [hunk 1]
```diff
@@ -9,8 +9,8 @@
         "php": ">=5.6,<8.0-dev",
         "ext-openssl": "*",
         "simplesamlphp/saml2": "3.2.*",
-        "symfony/dependency-injection": ">=2.7,<4",
-        "symfony/framework-bundle": ">=2.7,<4",
+        "symfony/dependency-injection": ">=2.7,<4 || ~4.3",
```
Comment (MKodde) [inline composer.json]: Do you have a good reason why you would only allow SF 4.3 and up? 
Are you certain the bundle still works with Symfony 2.7? You could include that version in the test stages. But I think it's even better to simply drop support for that version. I know of no Stepup application that uses Symfony 2.7.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,8 @@
+# 4.1.9
+**New feature**
+* Compatible with sf4
+* Classname alias support for sf4 auto-wiring
```
Comment (MKodde) [inline CHANGELOG.md]: New feature*s*

**Comment by MKodde** [comment]
@pablothedude and @epinxteren I've changed the composer.json slightly. Sharpening the Symfony requirements. Any feedback is welcome. If none is provided, I suggest to merge this and tag the 4.1.9 release!

**Comment by MKodde** [comment]
Note the PHP 7 build fails on the missing stepup-saml-bundle release that correctly requires the SF-di bundle

**Comment by MKodde** [comment]
> The only concern I have is that not all dependencies could be resolved. We're not able to detect this because the Symfony integration doesn't seem to have any test coverage.
> `src/Metadata/MetadataFactory.php:58` uses `\Symfony\Component\Templating\EngineInterface` which couldn't be found.

Thanks for spotting this. Fixed this by requireing `symfony/templating`, I'm not sure if this could ever have worked in SF 3.4 installations. As I could not find any dependencies on the templating dependency in any of our dependencies. Only in the dev-dependencies of the framework, twig and security bundles..

> I was wondering if some of the components needed to be retrieved with composer to support Symfony 4 but is seems that all components used are resolved with the Symfony framework bundle.

Thanks for checking this out :male_detective: 

> Personally I would be a an advocate to drop the Symfony 3 support completely and start a major 5 release.

Yes, we must certainly discuss this with SURFnet!

**Comment by MKodde** [comment]
@pablothedude after your review I've changed some things:

1. Require Symfony templating 
2. Pin the 3.4 versions to known safe versions
3. Update the README.md

Please review those changes and put your token of approval once satisfied.

**Comment by pablothedude** [comment]
@MKodde thanks for addressing the raised concerns and opening up a ticket to discuss the roadmap any further.
https://www.pivotaltracker.com/n/projects/1163646/stories/169838423

**Pull request merged**